src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -100,39 +104,169 @@
+
+    /**
+     * Wake up a nearby sleeping guard
+     *
+     * @return
+     */
+    private IAIState wakeUpGuard()
+    {
+        if (sleepingGuard.get() == null || !(sleepingGuard.get().getCitizenJobHandler().getColonyJob() instanceof AbstractJobGuard) || !sleepingGuard.get()
+                                                                                                                                          .getCitizenJobHandler()
+                                                                                                                                          .getColonyJob(AbstractJobGuard.class)
+                                                                                                                                          .isAsleep())
+        {
+            return DECIDE;
+        }
+
+        wakeTimer++;
+        // Wait 1 sec
+        if (wakeTimer == 1)
+        {
+            return getState();
+        }
+
+        // Move into range
+        if (BlockPosUtil.getDistanceSquared(sleepingGuard.get().getPosition(), worker.getPosition()) > 4 && wakeTimer <= 10)
+        {
+            worker.getNavigator().moveToEntityLiving(sleepingGuard.get(), getCombatMovementSpeed());
+        }
+        else
+        {
+            worker.swingArm(EnumHand.OFF_HAND);
+            sleepingGuard.get().attackEntityFrom(new DamageSource("wakeywakey").setDamageBypassesArmor(), 1);
+            sleepingGuard.get().setRevengeTarget(worker);
+            return DECIDE;
+        }
+
+        return getState();
+    }
+
+    /**
+     * Whether the guard should fall asleep.
+     *
+     * @return
+     */
+    private boolean shouldSleep()
+    {
+        if (worker.getRevengeTarget() != null || target != null)
+        {
+            return false;
+        }
+
+        if (worker.getRandom().nextInt(10) == 5)
+        {
+            sleepTimer = worker.getRandom().nextInt(500) + 2500;
+
+            final Entity entity = new SittingEntity(world, worker.posX, worker.posY - 1, worker.posZ, sleepTimer);
+            worker.startRiding(entity);
+            world.spawnEntity(entity);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Emits sleeping particles and regens hp when asleep
+     */
+    private IAIState sleepParticles()
+    {
+        MineColonies.getNetwork().sendToAllTracking(new SleepingParticleMessage(worker.posX, worker.posY, worker.posZ), worker);
+
+        if (worker.getHealth() < worker.getMaxHealth())
+        {
+            worker.setHealth(worker.getHealth() + 1);
+        }
+
+        return null;
+    }
+
+    /**
+     * Sleep activity
+     */
+    private IAIState sleep()
+    {
+        if (worker.getRevengeTarget() != null || (sleepTimer -= getTickRate()) < 0)
+        {
+            resetTarget();
+            worker.setRevengeTarget(null);
+            worker.dismountRidingEntity();
+            worker.setPosition(worker.posX, worker.posY + 1, worker.posZ);
```

Comment:
Might clean up the entity already here?

Reply:
hm not sure if its needed, it cleans itself up right away already